### PR TITLE
Add first draft of new, concise mentorship README

### PR DESCRIPTION
**Pull request by jjhampton**
@hollomancer @Exupery @rickr 

#### User Story: 
"As a prospective or new mentor/mentee, I can quickly read a one-page document that explains how Operation Code's Mentorship program works, and how I can start to participate."

For some time, the mentorship team has talked about creating a sharable, concise, one-page, ELI5 Mentorship doc that sums up what we're currently doing with the program in easy-to-read manner.  The Mentorship README could be a good candidate for this.

The [Guidebook](https://github.com/jjhampton/documents/blob/master/mentorship/GUIDEBOOK.md) has some great content, but it's a bit outdated, and worse, it's a 5-10 page novel.  We need something we can share with prospective/new mentors that sums it all up, and allows people to read it and then jump right in.  

I'm often asked by people on Slack interested in helping out, and I don't always have time to type out exactly how the current process works (mentor service requests, squads, etc).  Sharing a link like this can help with that.

Main goals: 
* Sum up how Mentor Service Requests work w/ Airtable
* Sum up the general idea behind Squads (this can be expanded to a separate page if we move forward with the idea of repurposing the Squads behind one of the platforms like Codecademy, FCC, etc.)

Open to any feedback.

### CODE REVIEW

File: mentorship/README.md [hunk 1]
```diff
@@ -2,58 +2,34 @@
...
 The Operation Code mentorship program serves to guide our community of service members and their families in their journey of learning to build software and starting a successful career in the tech industry.
 
-# Onboarding
-
-To join Operation Code as a mentor please sign up at https://operationcode.org/join. You'll receive an email inviting you to our slack community. Once you join introduce yourself, to include what technologies you're familiar with. You should also receive an invite to our mentor channel. 
-
-Once you're in slack please fill out http://op.co.de/mentor-enrollment. You should then receive an invite to our mentors-internal room.
-
 # Programs
+We offer two main mentorship platforms - [Mentor Service Requests](#mentor-service-requests), and [Squads](#squads).
```
Comment (Exupery) [inline mentorship/README.md]: I'm not sure if here or elsewhere on the page is the best place for it but I think _somewhere_ there should be a (brief) mention that another aspect of being a mentor is the more informal/ad-hoc assistance provided by participating in the various Slack channels.

File: mentorship/README.md [hunk 2]
```diff
@@ -2,58 +2,34 @@
...
 
-## 1 on 1
-
-When you sign up as a mentor you'll identify a few key areas you're able to help in. We'll use these areas to match you up with members who are requesting a mentor. When a memeber requests a mentor they'll be matched with one of our mentors. If you match the areas of interest for the mentee you'll be paired up and notified. 
-
-Please refer to the [Mentor Guidebook](https://github.com/OperationCode/documents/blob/master/mentorship/GUIDEBOOK.md) for more details on how to work with your mentee.
```
Comment (Exupery) [inline mentorship/README.md]: I'm not sure it's best to delete the reference/link to the guidebook. A succint readme is certainly valuable but it's important to let people know where to go for more information/details.

**N.B.** I do think this is an unusual spot for it though, it should either be in the opening paragraph or at the very bottom of the page.

Comment (hollomancer) [inline mentorship/README.md]: I agree with this.

**Comment by jjhampton** [comment]
@Hiren-Mehta @hollomancer @Exupery 
I added some changes to the README in response to your PR suggestions: 
* Remove references to Squads - none are currently active
* Mention informal/ad-hoc assistance that mentors provide via Slack / OC Community Discourse
* Add reference to Mentor Guidebook
